node/src/blockchain/test_utils.rs

```diff
@@ -132,86 +201,37 @@ impl BlockchainInterfaceMock {
             .push(result);
         self
     }
-}
-
-impl BlockchainInterface for BlockchainInterfaceMock {
-    fn contract_address(&self) -> Address {
-        self.contract_address_results.borrow_mut().remove(0)
-    }
 
-    fn retrieve_transactions(
-        &self,
-        start_block: u64,
-        recipient: &Wallet,
-    ) -> Result<RetrievedBlockchainTransactions, BlockchainError> {
-        self.retrieve_transactions_parameters
-            .lock()
-            .unwrap()
-            .push((start_block, recipient.clone()));
-        self.retrieve_transactions_results.borrow_mut().remove(0)
-    }
-
-    fn send_transaction<'b>(
-        &self,
-        inputs: SendTransactionInputs,
-    ) -> Result<(H256, SystemTime), BlockchainTransactionError> {
-        self.send_transaction_parameters
-            .lock()
-            .unwrap()
-            .push(inputs.abstract_for_assertions());
-        self.send_transaction_results.borrow_mut().remove(0)
-    }
-
-    fn get_eth_balance(&self, _address: &Wallet) -> Balance {
-        unimplemented!()
-    }
-
-    fn get_token_balance(&self, _address: &Wallet) -> Balance {
-        unimplemented!()
-    }
-
-    fn get_transaction_count(&self, wallet: &Wallet) -> Nonce {
-        self.get_transaction_count_parameters
-            .lock()
-            .unwrap()
-            .push(wallet.clone());
-        self.get_transaction_count_results.borrow_mut().remove(0)
-    }
-
-    fn get_transaction_receipt(&self, hash: H256) -> Receipt {
-        self.get_transaction_receipt_params
-            .lock()
-            .unwrap()
-            .push(hash);
-        self.get_transaction_receipt_results.borrow_mut().remove(0)
-    }
-
-    fn send_transaction_tools<'a>(
-        &'a self,
-        _fingerprint_request_recipient: &'a Recipient<PendingPayableFingerprint>,
-    ) -> Box<dyn SendTransactionToolsWrapper + 'a> {
-        self.send_transaction_tools_results.borrow_mut().remove(0)
+    pub fn get_block_number_result(self, result: LatestBlockNumber) -> Self {
+        self.get_block_number_results.borrow_mut().push(result);
+        self
     }
 }
 
 #[derive(Debug, Default, Clone)]
 pub struct TestTransport {
     asserted: usize,
-    requests: Rc<RefCell<Vec<(String, Vec<rpc::Value>)>>>,
-    responses: Rc<RefCell<VecDeque<rpc::Value>>>,
+    prepare_params: Arc<Mutex<Vec<(String, Vec<rpc::Value>)>>>,
+    send_params: Arc<Mutex<Vec<(RequestId, rpc::Call)>>>,
+    send_results: RefCell<VecDeque<rpc::Value>>,
+    send_batch_params: Arc<Mutex<Vec<Vec<(RequestId, rpc::Call)>>>>,
+    send_batch_results: RefCell<Vec<Vec<Result<rpc::Value, web3::Error>>>>,
+    //to check if we hold a true descendant of a certain initial instance
+    reference_counter_opt: Option<Arc<()>>,
```

Comment:
You probably don't need the two last lines. I implemented them but your version of the codebase doesn't need it.

It would stay idle for a while and it's better not to add it in now.